<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="4dof_URDF_9">
  <link name="world"/>
  <joint name="world_fixed_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
  </joint>
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="2.50508782488874 -0.711929863363831 -0.145140628319168"
        rpy="0 0 0" />
      <mass
        value="27.1336363085733" />
      <inertia
        ixx="0.608776925538367"
        ixy="1.82067856700149E-05"
        ixz="0.000271770918421296"
        iyy="0.676028748203949"
        iyz="-0.0281581607742424"
        izz="0.494218554478972" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.2 0.2 0.2 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="link_1">
    <inertial>
      <origin
        xyz="0 -0.130581484630826 0"
        rpy="0 0 0" />
      <mass
        value="2.98648132353624" />
      <inertia
        ixx="0.0240954782781149"
        ixy="3.61526161659195E-18"
        ixz="-1.77809156287623E-17"
        iyy="0.0105603189198943"
        iyz="-1.10023536061408E-17"
        izz="0.0230973900404768" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_0"
    type="revolute">
    <origin
      xyz="2.505 0.16781 -0.32162"
      rpy="1.5708 0 1.8311" />
    <parent
      link="base_link" />
    <child
      link="link_1" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.0"
      upper="1.0"
      effort="10"
      velocity="1.0" />
  </joint>
  <link
    name="link_2">
    <inertial>
      <origin
        xyz="-3.33066907387547E-16 0.0984013087850582 -2.40747498336802E-06"
        rpy="0 0 0" />
      <mass
        value="3.19882101039737" />
      <inertia
        ixx="0.0587854336457432"
        ixy="9.80118763926896E-17"
        ixz="-1.38777878078145E-17"
        iyy="0.016602340998814"
        iyz="1.33143591161565E-06"
        izz="0.0469634995041714" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_1"
    type="revolute">
    <origin
      xyz="0 -0.075 0"
      rpy="3.1416 0 -2.3253" />
    <parent
      link="link_1" />
    <child
      link="link_2" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1.0"
      upper="1.0"
      effort="10"
      velocity="1.0" />
  </joint>
  <link
    name="link_3">
    <inertial>
      <origin
        xyz="-0.152052929759594 6.93889390390723E-16 9.99999993922529E-08"
        rpy="0 0 0" />
      <mass
        value="1.34386283305885" />
      <inertia
        ixx="0.0291840488649348"
        ixy="1.97324795392362E-16"
        ixz="-3.05745012640912E-16"
        iyy="0.0729774407887769"
        iyz="1.46150452851046E-16"
        izz="0.0654377858813055" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_2"
    type="revolute">
    <origin
      xyz="0 0.25 0"
      rpy="3.1416 0 0.74466" />
    <parent
      link="link_2" />
    <child
      link="link_3" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1.0"
      upper="1.0"
      effort="10"
      velocity="1.0" />
  </joint>
  <link
    name="link_4">
    <inertial>
      <origin
        xyz="-8.88178419700125E-16 -7.21644966006352E-16 0.193808212195247"
        rpy="0 0 0" />
      <mass
        value="1.98027433388231" />
      <inertia
        ixx="0.0555349981058619"
        ixy="8.35919874986324E-17"
        ixz="9.97465998686664E-17"
        iyy="0.061090198105862"
        iyz="1.60245081093358E-16"
        izz="0.0198527234502471" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_4.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_4.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_3"
    type="prismatic">
    <origin
      xyz="-0.24416 0 0"
      rpy="3.1394 1.5708 -0.0025436" />
    <parent
      link="link_3" />
    <child
      link="link_4" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="10"
      velocity="1.0" />
  </joint>
  <link
    name="link_5">
    <inertial>
      <origin
        xyz="-6.66133814775094E-16 2.22044604925031E-16 0.0373920405725039"
        rpy="0 0 0" />
      <mass
        value="0.059452141686697" />
      <inertia
        ixx="2.26189174361307E-05"
        ixy="2.90108784434598E-20"
        ixz="-1.90582413132218E-20"
        iyy="2.88569332724916E-05"
        iyz="2.07523072077304E-20"
        izz="3.61664641457739E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_5.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 0.168627450980392 0.168627450980392 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_5.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_4"
    type="revolute">
    <origin
      xyz="0 0 0.33409"
      rpy="0 0 -2.361" />
    <parent
      link="link_4" />
    <child
      link="link_5" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="10"
      velocity="1.0" />
  </joint>
  <link
    name="link_6">
    <inertial>
      <origin
        xyz="0.0155076727184624 4.44089209850063E-15 0.0481870569436075"
        rpy="0 0 0" />
      <mass
        value="0.040102943291367" />
      <inertia
        ixx="4.15706671935077E-05"
        ixy="5.70688448212585E-20"
        ixz="-9.46943385826672E-06"
        iyy="4.55830549069341E-05"
        iyz="1.40501590125807E-19"
        izz="6.18763426294392E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_6.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.2 0.2 0.2 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_6.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_5"
    type="revolute">
    <origin
      xyz="0.041194 0 0.051181"
      rpy="0 -0.086999 3.1416" />
    <parent
      link="link_5" />
    <child
      link="link_6" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.0"
      upper="1.0"
      effort="10"
      velocity="1.0" />
  </joint>
  <link
    name="link_7">
    <inertial>
      <origin
        xyz="0.0155076727184587 -4.21884749357559E-15 0.0481870569436035"
        rpy="0 0 0" />
      <mass
        value="0.0401029432913669" />
      <inertia
        ixx="4.15706671935076E-05"
        ixy="-3.18696146404431E-20"
        ixz="-9.46943385826668E-06"
        iyy="4.55830549069339E-05"
        iyz="-1.14544662503893E-19"
        izz="6.18763426294386E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_7.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.2 0.2 0.2 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/venus/ros2_ws/src/cli_spawn/meshes/link_7.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_6"
    type="revolute">
    <origin
      xyz="-0.041194 0 0.051181"
      rpy="0 -0.31849 0" />
    <parent
      link="link_5" />
    <child
      link="link_7" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.0"
      upper="1.0"
      effort="10"
      velocity="1.0" />
  </joint>
  <ros2_control name="GazeboSimSystem" type="system">
    <hardware>
      <plugin>gz_ros2_control/GazeboSimSystem</plugin>
    </hardware>
    <joint name="joint_0">
      <command_interface name="position">
        <param name="min">-1.0</param>
        <param name="max">1.0</param>
      </command_interface>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
    </joint>
    <joint name="joint_1">
      <command_interface name="position">
        <param name="min">-1.0</param>
        <param name="max">1.0</param>
      </command_interface>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
    </joint>
    <joint name="joint_2">
      <command_interface name="position">
        <param name="min">-1.0</param>
        <param name="max">1.0</param>
      </command_interface>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
    </joint>
    <joint name="joint_3">
      <command_interface name="position">
        <param name="min">-1.0</param>
        <param name="max">1.0</param>
      </command_interface>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
    </joint>
    <joint name="joint_4">
      <command_interface name="position">
        <param name="min">-1.0</param>
        <param name="max">1.0</param>
      </command_interface>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
    </joint>
    <joint name="joint_5">
      <command_interface name="position">
        <param name="min">-1.0</param>
        <param name="max">1.0</param>
      </command_interface>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
    </joint>
    <joint name="joint_6">
      <command_interface name="position">
        <param name="min">-1.0</param>
        <param name="max">1.0</param>
      </command_interface>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
    </joint>
  </ros2_control>
  <gazebo>
    <plugin filename="libgz_ros2_control-system.so" name="gz_ros2_control::GazeboSimROS2ControlPlugin">
      <parameters>/home/venus/ros2_ws/src/cli_spawn/config/myrobot_control.yaml</parameters>
    </plugin>
  </gazebo>
</robot>
